src/lib/gundb/UserStorageClass.js

```diff
@@ -1332,7 +1332,10 @@ export class UserStorage {
       profile.smallAvatar = await resizeImage(profile.avatar, 50)
     }
 
-    const fieldsToSave = keys(this.profileSettings).filter(key => profile[key])
+    const fieldsToSave = keys(this.profileSettings)
```

Comment:
the following workaround is very narrow. let's make it more universal using lodash's `_.defaults`: